src/include/cursor_inline.h

```diff
@@ -74,6 +74,31 @@ __cursor_novalue(WT_CURSOR *cursor)
     F_CLR(cursor, WT_CURSTD_VALUE_INT);
 }
 
+/*
+ * __wt_cursor_bound_reset --
+ *     Clear any bounds on the cursor if they are set.
+ */
+static inline void
+__wt_cursor_bound_reset(WT_CURSOR *cursor)
+{
+    WT_SESSION_IMPL *session;
+
+    session = CUR2S(cursor);
+
+    /* Clear bounds if they are set. */
```

Comment:
Could you add an assert in here that it's only ever called when `API_FIRST_ENTRY` is true?

Reply:
Unfortunately we need to call this function explicitly before caching the cursor into the pool of available cursors on close.